bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -379,10 +378,10 @@ public synchronized void playlistRemove(int playlistID, int position) {
      * @param favoriteTitle the title of the favorite
      * @return the ({@link KodiFavorite}) with the given title
      */
-    @Nullable
-    public KodiFavorite getFavorite(final String favoriteTitle) {
+    public @Nullable KodiFavorite getFavorite(final String favoriteTitle) {
         for (KodiFavorite favorite : getFavorites()) {
-            if (StringUtils.equalsIgnoreCase(favorite.getTitle(), favoriteTitle)) {
+            String title = favorite.getTitle();
+            if (title != null && title.equals(favoriteTitle)) {
```

Comment:
here too

Reply:
Unless you are sure that case doesn't need to be ignored, I suggest keeping it case-insensitive.
```suggestion
            if (title != null && title.equalsIgnoreCase(favoriteTitle)) {
```

Reply:
Looking at the annotations `favoriteTitle` seems to be non null, so inverting would help you to get rid of the `null`-check